### PR TITLE
Lect2

### PR DESCRIPTION
**Pull request by Jagrmi-C**

### CODE REVIEW

File: median/main.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	"fmt"
+	"log"
+	"sort"
+)
+
+func median(intsArr []int) float64 {
+	var answer float64
+	val := len(intsArr) / 2
+	log.Println("Half value: ", val)
+	if len(intsArr)%2 == 1 {
```
Comment (ikamornikau) [inline median/main.go:13]: This `if ... else` clause is too long. I would prefer something like:
```
if len(intsArr)%2 == 1 {
        log...
        return float64(intsArr[val])
}

log...
return ...
```

File: median/main.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		answer = (float64(intsArr[val-1]) + float64(intsArr[val])) / 2
+		log.Println("Answer:", answer)
+	}
+	return answer
+}
+
+func main() {
+	intsArr := []int{7, 2, 4, 8, 1, 2, 9}
+	// intsArr := []int{7, 2, 4, 8, 1, 9}
+	sort.Ints(intsArr)
```
Comment (ikamornikau) [inline median/main.go]: If I added some `file.go` to the package and wanted to use `median()` func, I could miss the fact that the slice (`intsArr []int` - `median()` parameter) must be sorted. Please move you sorting inside `median()` func.

File: square/main.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package main
+
+import "fmt"
+
+// Point is position in space
```
Comment (ikamornikau) [inline square/main.go]: Please make your comment more clear:
```
// Point type represents a point in Cartesian coordinate system
```

File: square/main.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+type Point struct {
+	x, y int
+}
+
+// Square is equilateral rectangle
+type Square struct {
+	start Point
+	a     uint
```
Comment (ikamornikau) [inline square/main.go]: Please add a comment to describe `a` variable.

File: square/main.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+func (s Square) Perimeter() int {
+	return int(s.a) * 4
+}
+
+// Area - calculate area for Square
+func (s Square) Area() int {
+	return int(s.a) * int(s.a)
+}
+
+// End - Point after made ...
```
Comment (ikamornikau) [inline square/main.go]: Please make this comment more clear.

File: square/main.go [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+func (s Square) End() Point {
+	newX := s.start.x + int(s.a)
+	newY := s.start.y - int(s.a)
+	endPoint := Point{newX, newY}
```
Comment (ikamornikau) [inline square/main.go]: I would do something like:
```
return Point{
        x: s.start.x + int(s.a),
        y: s.start.y - int(s.a),
}
```

File: median/main.go [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+	} else {
+		log.Println("%2 == 0")
+		answer = (float64(intsArr[val-1]) + float64(intsArr[val])) / 2
+		log.Println("Answer:", answer)
+	}
+	return answer
+}
+
+func main() {
+	intsArr := []int{7, 2, 4, 8, 1, 2, 9}
```
Comment (ikamornikau) [inline median/main.go]: Please add one case for odd amount of slice elements.

**Review by ikamornikau — CHANGES_REQUESTED** [review]
Please address the requested changes.

File: median/main.go [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+func median(intsArr []int) float64 {
+	var answer float64
+	sort.Ints(intsArr)
+	log.Println("Sorted list:", intsArr)
+	val := len(intsArr) / 2
+	log.Println("Half value: ", val)
+	if len(intsArr)%2 == 1 {
+		log.Println("%2 == 1")
+		answer = float64(intsArr[val])
+		log.Println("Answer:", answer)
```
Comment (VolhaVetrava) [inline median/main.go]: You're already printing the answer in the main function, so you don't really need to do it here as well, and then the variable `answer` is redundant and you can just return the value.

File: median/main.go [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+		return answer
+	}
+
+	log.Println("%2 == 0")
+	answer = (float64(intsArr[val-1]) + float64(intsArr[val])) / 2
+	log.Println("Answer:", answer)
```
Comment (VolhaVetrava) [inline median/main.go]: Same as above

File: median/main.go [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+	val := len(intsArr) / 2
+	log.Println("Half value: ", val)
+	if len(intsArr)%2 == 1 {
+		log.Println("%2 == 1")
+		answer = float64(intsArr[val])
+		return answer
+	}
+
+	log.Println("%2 == 0")
+	answer = (float64(intsArr[val-1]) + float64(intsArr[val])) / 2
```
Comment (VolhaVetrava) [inline median/main.go]: Variable answer is redundant, you can just write `return (float64(intsArr[val-1]) + float64(intsArr[val])) / 2`

File: median/main.go [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+func median(intsArr []int) float64 {
+	var answer float64
+	sort.Ints(intsArr)
+	log.Println("Sorted list:", intsArr)
+	val := len(intsArr) / 2
+	log.Println("Half value: ", val)
+	if len(intsArr)%2 == 1 {
+		log.Println("%2 == 1")
+		answer = float64(intsArr[val])
```
Comment (VolhaVetrava) [inline median/main.go]: 'Answer' is redundant, same as below

**Review by VolhaVetrava — CHANGES_REQUESTED** [review]

**Pull request closed without merging**